app/javascript/gobierto_data/webapp/components/SidebarDatasets.vue

```diff
@@ -0,0 +1,136 @@
+          :href="'/datos/' + item.attributes.slug"
+          class="gobierto-data-sidebar-datasets-name"
+          @click.prevent="nav(item.attributes.slug, item.attributes.name, index)"
+        >{{ item.attributes.name }}
+        </a>
+        <div
+          v-show="toggle === index"
+          class="gobierto-data-sidebar-datasets-container-columns"
+        >
+          <div v-if="showLess">
+            <template v-for="(column, i) in lessColumns">
+              <span
+                :key="i"
+                class="gobierto-data-sidebar-datasets-links-columns"
+              >
+                {{ column }}
+              </span>
+            </template>
+          </div>
+          <div v-else>
+            <span
+              v-for="(column, i) in item.attributes.columns"
+              :key="i"
+              class="gobierto-data-sidebar-datasets-links-columns"
+            >
+              {{ i }}
+            </span>
+          </div>
+          <div v-if="Object.keys(item.attributes.columns).length > showMaxKeys">
+            <template
+              v-if="showLess"
+            >
+              <span
+                class="gobierto-data-sidebar-datasets-links-columns-see-more"
+                @click="showLess = false"
+              >
+                {{ labelshowAll }}
+              </span>
+            </template>
+            <template v-else>
+              <span
+                class="gobierto-data-sidebar-datasets-links-columns-see-more"
+                @click="showLess = true"
+              >
+                {{ labelshowLess }}
+              </span>
+            </template>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+import { store } from "./../../lib/store";
+export default {
+  name: "SidebarDatasets",
+  data() {
+    return {
+      labelSets: "",
+      labelQueries: "",
+      labelCategories: "",
+      labelSeeMore: "",
+      labelSeeLess: "",
+      listDatasets: [],
+      toggle: 0,
+      indexToggle: null,
+      showMaxKeys: 10,
+      showLess: true,
+      lessColumns: [],
+      items: store.state.datasets || []
+    }
+  },
+  created() {
+    this.labelSets = I18n.t("gobierto_data.projects.sets")
+    this.labelQueries = I18n.t("gobierto_data.projects.queries")
+    this.labelCategories = I18n.t("gobierto_data.projects.categories")
+    this.labelshowAll = I18n.t("gobierto_data.projects.showAll")
+    this.labelshowLess = I18n.t("gobierto_data.projects.showLess")
+    this.orderDatasets()
+  },
+  methods: {
+    orderDatasets() {
+      const sortDatasets = this.items
+      const allDatasets = sortDatasets.sort((a, b) => a.attributes.name.localeCompare(b.attributes.name));
+      let slug = this.$route.params.id
+      const indexToggle = allDatasets.findIndex(dataset => dataset.attributes.slug == slug)
+      this.toggle = indexToggle
+      if (this.toggle === -1) {
+        this.toggle = 0
+        slug = allDatasets[0].attributes.slug
```

Comment:
Provide fallbacks when you do this kind of things. For example:
```suggestion
        slug = (((allDatasets || [])[0] || {}).attributes || {}).slug
```
That way never will output any error. But if you assume some things. you may do some simpler: 
```suggestion
        slug = allDatasets.length ? allDatasets[0].attributes.slug : ''
```